### PR TITLE
[fix bug 1411077] Whatsnew page for Firefox 57 Quantum

### PR DESCRIPTION
**Pull request by craigcook**
## Description
Firefox 57 will have a custom whatsnew page with a few variations. One version will be a letter from Chris Beard translated in 8 primary languages, and the other is a minimal baseline for all other locales.

## Issue / Bugzilla link
[Baseline page for all locales](https://bugzilla.mozilla.org/show_bug.cgi?id=1411077)
[Letter variant for key locales](https://bugzilla.mozilla.org/show_bug.cgi?id=1411076)

## Testing
[Demo](https://bedrock-demo-craigcook.us-west.moz.works/firefox/57.0/whatsnew/)

### CODE REVIEW

**Comment by craigcook** [comment]
@peiying2  I'll be adding locale-specific templates for the 8 primary locales that will show the letter and they'll be excluded from the regular l10n process so we're only concerned with `whatsnew-57.html` for now (lang file is `firefox/whatsnew_57`).

The baseline page has a total of five new strings, all short and simple, so hopefully it won't present too much of a burden for localizers. Let me know if you spot any issues. Thanks!

**Comment by peiying2** [comment]
@craigcook the file is prepared. We need to know the demo URL for this page so localizers can verify their work in context.

File: media/css/firefox/whatsnew/whatsnew-57.scss [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+        h2 {
+            @include font-size-level3;
+            font-weight: bold;
+        }
+
+        .tagline {
+            margin: 20px 0 40px;
+
+            br {
```
Comment (flodolo) [inline media/css/firefox/whatsnew/whatsnew-57.scss]: Need to remove this (CC @MikkCZ)
https://bedrock-demo-craigcook.us-west.moz.works/cs/firefox/57.0/whatsnew/

Comment (flodolo) [reply media/css/firefox/whatsnew/whatsnew-57.scss]: For reference, this is really bad. You would end up removing a perfectly fine `<br>`, and breaking the sentence (e.g. missing a space after a period or comma).

File: bedrock/firefox/templates/firefox/new/quantum/base.html [hunk 2]
```diff
@@ -45,7 +45,7 @@
     {% block messages %}{% endblock %}
     <div class="content">
       <div class="header-logos">
-        <h2><a class="firefox" href="{{ url('firefox') }}">{{ high_res_img('logos/firefox/logo-quantum-wordmark-white.png', {'alt': 'Firefox', 'width': '133', 'height': '51'}) }}</a></h2>
+        <h2><a class="firefox" href="{{ url('firefox') }}">{{ high_res_img('logos/firefox/logo-quantum-wordmark-white.png', {'alt': 'Firefox', 'width': '133', 'height': '50'}) }}</a></h2>
```
Comment (craigcook) [inline bedrock/firefox/templates/firefox/new/quantum/base.html]: I needed a slightly bigger version of this image but didn't want to add yet another dupe logo so I updated the original but had to tweak the size here. This was the only other place it was used so far, but if other new pages also rely on this asset they may need to adjust sizing.

**Comment by craigcook** [comment]
All locales have signed off on the letter translations so this is ready for code review.

File: media/css/firefox/whatsnew/whatsnew-57.scss [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        @media #{$mq-tall} {
+            min-height: 40vh;
+
+            .firefox-logo {
+                margin-top: 10vh;
+            }
+        }
+
+        @media (min-height: 800px) {
+            min-height: 60vh;
```
Comment (alexgibson) [inline media/css/firefox/whatsnew/whatsnew-57.scss]: On my laptop this value feels kinda big, with most of the letter off below screen. `40vh` feels about right though. Could this do with an additional check for a really large width?

File: bedrock/firefox/templates/firefox/new/quantum/base.html (hunk 2 above)
Comment (bigdickbrinson67) [reply bedrock/firefox/templates/firefox/new/quantum/base.html]: Yes

File: bedrock/firefox/views.py [hunk 4]
```diff
@@ -340,20 +340,29 @@ def show_54_whatsnew(version):
...
             oldversion = Version(oldversion)
     except ValueError:
         return False
 
-    v56 = Version('56.0')
+    v57 = Version('57.0')
 
     if oldversion:
-        return version >= v56 and oldversion < v56
+        return version >= v57 and oldversion < v57
```
[the remark this answers is not in the document]
Comment (craigcook) [reply bedrock/firefox/views.py]: I just copied the old tests. I fully expected to get this wrong.
noidea-jingle

**Comment by craigcook** [comment]
Updated! I was clearly overthinking the view and tests. We don't need to do anything special for oldversion and the v56 logic should still apply for v56, so I've reverted that stuff and simplified the 57 bits.

**Comment by craigcook** [comment]
Also of note, flod has expressed concerns about locale-specific templates without an active lang file not loading properly in production mode, so we should test that thoroughly (I guess there were issues with this before but maybe that was a bug that got fixed). It all works as it should for me locally with DEV=False but I don't know if I fully trust that on a local environment. Maybe we can set a demo server to production mode to verify?

**Pull request merged**